src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -65,5 +67,34 @@ public IActionResult GetInstanceInfo()
         {
             return Ok(_instanceManager.GetInstanceInfo());
         }
+
+        /// <summary>
+        /// This endpoint generates a temporary x-ms-site-restricted-token for core tool
+        /// to access KuduLite zipdeploy endpoint in Linux Consumption
+        /// </summary>
+        /// <returns>
+        /// 200 on token generated
+        /// 400 on non-Linux container environment
+        /// 404 on WEBSITE_AUTH_ENCRYPTION_KEY is not set
+        /// </returns>
+        [HttpGet]
+        [Route("admin/getsitetoken")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public IActionResult GetSiteToken()
+        {
+            if (!_environment.IsLinuxContainerEnvironment())
+            {
+                return BadRequest("Endpoint is only available when running in Linux Container");
+            }
+
+            string websiteEncryptionKey = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey);
+            if (string.IsNullOrEmpty(websiteEncryptionKey))
```

Comment:
Rather than duplicate all this logic, can you just use `SimpleWebTokenHelper.CreateToken(DateTime.UtcNow.AddMinutes(5))` which we use elsewhere for generating site tokens?